### PR TITLE
Maintenance loot table revamp

### PR DESCRIPTION
**Pull request by ghost**
Adds and removes various maintenance loot items, and converts loot tables to an easier to modify nested list format.

Maintenance loot has been divided into junk, common, and uncommon lists, each with its own established power level. Each category also has its own weight, determining how often that category is rolled. This allows for you to add/remove items from any category without affecting how often the category is selected. Previously, adding or removing any item affected the spawn rates of every other item.

Also, I made a special loot category: oddity, with a 1/10,000 spawn chance (spawns around 1.8% of rounds). It contains strange items, such as a real blood rune. This category is for creating memorable in-game stories.

Concerns:
I tried to keep the loot tame, but I may have made the average loot power level too high. This can be fixed by adjusting the useful category weights downward. 
Also, I may have bloated the loot table up too much. I'm not sure if having lots of loot variety is important enough to justify the big table I made. I can thin it out as needed.

:cl:
balance: New junk is being found in the station's maintenance tunnels.
/:cl:

### CODE REVIEW

**Comment by wesoda25** [comment]
For uncommon loot I recommend:

Random Present (potential for greatness - or nothing)
Collectable Hat
I'd recommend a random traitor item, but I don't think thats what you want with this category
Maybe unused AI Modules? (in the same vein, many unused boards would work well, such as donksoft)

idk ill update if I think of more

**Comment by py01** [comment]
Further additions can be handled in a different PR - and I don't want the list to get too bloated with random stuff.

**Comment by carlarctg** [comment]
image

why did you take sunglasses out entirely?
i absolutely love the idea of maint actually having useful weapons, but maybe a stunprod is a bit too much. maybe only make it spawn prods without cells instead?
should probably add some more useful items to the medicine section. perhaps a morphine bottle and a syriniver syringe
maybe increase the weight of grey bull, this very biased but i feel its use should be encouraged so it's less of a powergamer-only thing

**Comment by necromanceranne** [comment]
I'm okay with shades being more and more rare overall but I think this is only highlighting how we do need to get rid of that and soonish.

**Comment by py01** [comment]
I'm specifically not adding everything right now, because it was a rabbit hole of new ideas that went forever, I had to stop and actually PR at some point.

The stunprod spawns without a cell. Considering how easy they are to make, I'm not concered. (15 cables, 1 igniter, (buyable from public vendor in tool storage) 1 metal rod

**Comment by wesoda25** [comment]
Adding what formerly was there won't make a run on PR, it'll just keep content in game. I get not wanting a brainstorming session for other things but the bare minimum is what was previously there.

File: code/_globalvars/lists/maintenance_loot.dm [hunk 1]
```diff
@@ -1,111 +1,250 @@
...
+		/obj/item/clothing/suit/armor/vest/old  = 1,
+		/obj/item/clothing/head/helmet/old = 1,
+		/obj/item/clothing/mask/muzzle = 1,
+		/obj/item/clothing/ears/earmuffs = 1,
+		/obj/item/clothing/gloves/color/black = 1,
+		) = 1,
+
+	list(//construction and crafting
+		/obj/item/stock_parts/cell/high = 1,
+		/obj/item/stack/sheet/mineral/wood/fifty = 1, //todo:replace with smaller amount
```
Comment (tralezab) [inline code/_globalvars/lists/maintenance_loot.dm]: you're revamping the loot table, either do the comment or remove it

**Comment by AnonymousNow** [comment]
Adding little systems like this is a wonderful thing.

I know that you're not adding more content in this PR, or as you've said, but ever since back when some traitor items had identical sprites to maintenance kipple (the Chameleon Projector shared a sprite with a simple timer, IIRC, for example), I was fond of the idea of every maintenance item that had a counterpart had a remarkably out there - like, one in a thousand - chance to be its counterpart. The day of that idea might be past now, but still, it's nice to see the loot being reworked to be more interesting.

**Comment by Timonkeyn** [comment]
You removed maintenance pumpups

**Comment by py01** [comment]
Pump up line 157

**Comment by CPTANT** [comment]
Please add yellow gloves the drop table somewhere.

All static spawns got removed but nothing in the drop tables was altered.

**Comment by Twaticus** [comment]
when sunglasses were removed from static spawns, they were purposely left as random maint loot. why remove them now?

File: code/_globalvars/lists/maintenance_loot.dm [hunk 2]
```diff
@@ -1,111 +1,250 @@
...
+		/obj/item/assembly/mousetrap = 1,
+		/obj/item/assembly/prox_sensor = 1,
+		/obj/item/assembly/timer = 1,
+		/obj/item/assembly/igniter = 1,
+		/obj/item/assembly/health = 1,
+		/obj/item/stack/packageWrap = 1,
+		) = 1,
+
+	list(//medical and chemicals
+		/obj/item/reagent_containers/pill/floorpill = 1,
```
Comment (actioninja) [inline code/_globalvars/lists/maintenance_loot.dm]: floorpills are probably better suited to trash

File: code/_globalvars/lists/maintenance_loot.dm [hunk 3]
```diff
@@ -1,111 +1,250 @@
...
+		/obj/item/multitool = 1,
+		/obj/item/hatchet = 1,
+		/obj/item/roller = 1,
+		/obj/item/restraints/legcuffs/bola = 1,
+		/obj/item/restraints/handcuffs/cable = 1,
+		/obj/item/twohanded/spear = 1,
+		/obj/item/shield/riot/buckler = 1,
+		/obj/item/grenade/iedcasing = 1,
+		/obj/item/melee/baton/cattleprod = 1,
+		) = 1,
```
Comment (actioninja) [inline code/_globalvars/lists/maintenance_loot.dm]: these are all pretty toasty of items, have you tested how frequently these spawn?

**Review by actioninja — CHANGES_REQUESTED** [review]
This is definitely a pretty marked shift in making random maintenance loot higher value, which I'm not sure how I feel about.

The likelihood of specific stuff can be calculated, but I'd like some actual numbers on maint loot results with this pool.

**Comment by AnonymousNow** [comment]
Dang. Hope this gets revisited.

**Comment by py01** [comment]
I just fucked my git, fixing it

**Comment by AnonymousNow** [comment]
That was a fast revisit.

File: code/_globalvars/lists/maintenance_loot.dm (hunk 3 above)
Comment (py01) [reply code/_globalvars/lists/maintenance_loot.dm]: Each item in your marked section currently has a 1/600 probability of selection. (1/10 to select uncommon, 1/6 to select that tools list, 1/10 to select any given sub-item). For metastation, with ~180 pieces of total loot spawned, an average of 3 from this section, and 0.3 of each individual item will spawn, and an average of 18 items from the uncommon category overall.

Comment (py01) [reply code/_globalvars/lists/maintenance_loot.dm]: The three defines from lines 239-242 control the spawn rates of each main category. If you feel that the loot is too strong, I could move the uncommon selection rate down from 1/10 to 3/100, or whatever, it's fairly flexible and I'm not attached to any of the spawn rates.

Comment (actioninja) [reply code/_globalvars/lists/maintenance_loot.dm]: ~3 is a not entirely unreasonable number for that value. It can always be tweaked going forward.

**Comment by actioninja** [comment]
Overall I'd say I like this, but track some stats of what a generated loot pool looks like. Given the nature of RNG some are going to be hot but I still just want an idea of what this would actually look like in practice.

**Comment by py01** [comment]
In order to make balance assessments easier, I have made an item that "creates" sample loot pools from a given spawner, creating a list of all items "spawned" and how many, and a list of their probabilities. It works by repeatedly selecting items from the loot table and recording how many of each spawn.

I will look into putting a sample spawn table into a figure or something.

**Comment by py01** [comment]
I've made [lootdrop excel table](https://docs.google.com/spreadsheets/d/19p6DL53DlEh_8oW4tLYvRP2ddkw0plpDyggR7IhfkAc/edit#gid=2086844162). It contains a sample set of 180 items created, and a 50,000-set sample used to create the estimated probabilities of all items.

**Comment by actioninja** [comment]
That was exactly what I was looking for, great job on that.
Some thoughts from looking at an actual table:
trash items are not very common
manuals are very common
there's a lot of pump up

**Comment by py01** [comment]
-the trash list is selected 45% of the time (from the define weights) but this specific sample only rolled 33% trash, also the long-ish list giving individual trash pieces a low chance likely makes it seem worse than it is. The define can be changed if 45% avg is too low, I can cut the common rate which is also 45% which might be too high
-books cut because no one really reads them
-pump up avg spawn count for 180 sample set moved from 11.6 to 3.4

**Review by actioninja — DISMISSED** [review]
As previously stated this is a pretty big shift and is probably going to have a lot of balance issues on initial iterations, but it's an interesting change that hopefully could eventually mean a near total death of fixed maint loot spawns.
Tentatively approving this with the stipulation that if major problems creep up it will be fixed

**Comment by actioninja** [comment]
Conflict

**Comment by py01** [comment]
conflict was from shuriken added to maint loot by #47820, put the shuriken in uncommon tools to be polite. Probably a bit too strong, but it's going to be quite rare.

**Comment by actioninja** [comment]
time to rock n roll

**Comment by cacogen** [comment]
hope this doesn't ruin maint diving

**Pull request merged**